### PR TITLE
Add telemetry for `meltano config`

### LINKED ISSUE #6190: Add telemetry for `meltano config` command
Update `meltano.cli.config` to support new style snowplow telemetry. 

Spun out from https://github.com/orgs/meltano/discussions/392

### PR DESCRIPTION
**Pull request by pandemicsyn**
Add telemetry to the `config` cli command.

Like some of the other commands, the `config` command uses a click a command group. So I was able to initialize the tracker there and seed it with the cli context AND the plugin context. 

closes #6190

### CODE REVIEW

**Comment by aaronsteers** [comment]
@pandemicsyn - would you like to try to fit this in to `2.0.1`?

**Comment by aaronsteers** [comment]
@pandemicsyn - I'm gonna go ahead and start the release process. If it finishes before I publish, happy to include - but also totally fine if not.

**Comment by pandemicsyn** [comment]
@aaronsteers depends on how important config is, i think its done but it could use a once over from @pnadolny13 

I think this does cover most of the success cases, and because this command uses a command group - i was able to initialize all the important contexts early on - so even a failure should have the plugin context attached (as long as the plugin was valid). 

There are a few edge cases where I can't fire a failure event currently (like if you pass in to many args to `unset`) but we should get those when we implement exit events.

**Comment by pandemicsyn** [comment]
@aaronsteers IMHO this ones not critical and we already got the big ones merged `invoke` , `elt`, `run`, `add`, `install`. I'd think this one is fine to go out in 2.0.2.

**Comment by pandemicsyn** [comment]
Note: https://github.com/meltano/meltano/pull/6148 has a bunch of clean up that will impact this PR.

**Comment by pandemicsyn** [comment]
Refactored to work with #6148 , should be good to go.

**Review by cjohnhanson — CHANGES_REQUESTED** [review]

File: src/meltano/cli/config.py [hunk 1]
```diff
@@ -66,8 +84,13 @@ def config(  # noqa: WPS231
         if plugin_name == "meltano":
             plugin = None
         else:
+            tracker.track_command_event(CliEvent.started)
+            tracker.track_command_event(CliEvent.aborted)
             raise
 
+    tracker.add_contexts(PluginsTrackingContext([(plugin, None)]))
```
[the remark this answers is not in the document]
Comment (pandemicsyn) [reply src/meltano/cli/config.py]: https://github.com/meltano/meltano/pull/6137/commits/48070fe784bec08717b9c94f9b336d5de9ab6c85

**Review by cjohnhanson — APPROVED** [review]

**Review by edgarrmondragon — APPROVED** [review]

**Comment by pandemicsyn** [comment]
sorry for the "draft" -> "ready for review" spam. Intended to flip a different PR back to draft.

**Pull request merged**